src/main/scala/com/actian/spark_vector/colbuffer/time/TimeColumnBuffer.scala

```diff
@@ -18,67 +18,72 @@ package com.actian.spark_vector.colbuffer.time
+}
 
-    override def convert(nanos: Long, scale: Int): Long = TimeConversion.scaledTime(nanos, scale)
-  }
+private class TimeNZLZConverter extends TimeConversion.TimeConverter {
 
-  override protected def createConverter(): TimeConversion.TimeConverter = new TimeLZConverter()
+  override def convert(nanos: Long, scale: Int): Long = TimeConversion.scaledTime(nanos, scale)
 }
 
-private[colbuffer] trait TimeNZColumnBufferInstance extends TimeColumnBufferInstance {
+private class TimeTZConverter extends TimeConversion.TimeConverter {
+  private final val TimeMask = 0xFFFFFFFFFFFFF800L
 
-  protected def supportsNZColumnType(tpe: String, columnScale: Int, minScale: Int, maxScale: Int): Boolean =
-    (tpe.equalsIgnoreCase(TimeNZTypeId1) || tpe.equalsIgnoreCase(TimeNZTypeId2)) && minScale <= columnScale && columnScale <= maxScale
+  override def convert(nanos: Long, scale: Int): Long = (TimeConversion.scaledTime(nanos, scale) << 11) & TimeMask
+}
 
-  private class TimeNZConverter extends TimeConversion.TimeConverter {
+/** Builds a `ColumnBuffer` object for `time` (NZ, TZ, LZ) types. */
+private[colbuffer] object TimeColumnBuffer extends ColumnBufferBuilder {
+  private final val (nzlzIntScaleBounds, nzlzLongScaleBounds) = ((0, 4), (5, 9))
+  private final val (tzIntScaleBounds, tzLongScaleBounds) = ((0, 1), (2, 9))
 
-    override def convert(nanos: Long, scale: Int): Long = TimeConversion.scaledTime(nanos, scale)
+  private val buildNZPartial: PartialFunction[ColumnBufferBuildParams, TimeColumnBufferParams] = {
+    case p if p.tpe.equalsIgnoreCase(TimeNZTypeId1) || p.tpe.equalsIgnoreCase(TimeNZTypeId2) => TimeColumnBufferParams(p, new TimeNZLZConverter(), true)
   }
 
-  override protected def createConverter(): TimeConversion.TimeConverter = new TimeNZConverter()
-}
-
-private[colbuffer] trait TimeTZColumnBufferInstance extends TimeColumnBufferInstance {
+  private val buildLZPartial: PartialFunction[ColumnBufferBuildParams, TimeColumnBufferParams] = {
+    case p if p.tpe.equalsIgnoreCase(TimeLZTypeId) => TimeColumnBufferParams(p, new TimeNZLZConverter())
+  }
 
-  protected def supportsTZColumnType(tpe: String, columnScale: Int, minScale: Int, maxScale: Int): Boolean = {
-    tpe.equalsIgnoreCase(TimeTZTypeId) && minScale <= columnScale && columnScale <= maxScale
+  private val buildNZLZ: PartialFunction[ColumnBufferBuildParams, ColumnBuffer[_]] = (buildNZPartial orElse buildLZPartial) andThenPartial {
+    case nzlz if isInBounds(nzlz.cbParams.scale, nzlzIntScaleBounds) => new TimeIntColumnBuffer(nzlz)
+    case nzlz if isInBounds(nzlz.cbParams.scale, nzlzLongScaleBounds) => new TimeLongColumnBuffer(nzlz)
   }
 
-  private class TimeTZConverter extends TimeConversion.TimeConverter {
-    private final val TimeMask = 0xFFFFFFFFFFFFF800L
+  private val buildTZPartial: PartialFunction[ColumnBufferBuildParams, TimeColumnBufferParams] = {
+    case p if p.tpe.equalsIgnoreCase(TimeTZTypeId) => TimeColumnBufferParams(p, new TimeTZConverter())
```

Comment:
could we make sure when we create the ColumnBufferBuildParams that we convert the tpe to lower case and then just do equality checks instead of calling equalsIgnoreCase ?